src/Util/BatchRequest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Http\Client\Util;
+
+use Http\Client\BatchResult;
+use Http\Client\Exception;
+use Http\Client\Exception\BatchException;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Implements sending multiple request for client not supporting parallels requests
+ *
+ * Internally, use the sendRequest method
+ *
+ * Should be used with Http\Client\HttpPsrClient
+ *
+ * @author Joel Wurtz <jwurtz@jolicode.com>
+ */
+trait BatchRequest
+{
+    /**
+     * {@inheritdoc}
+     */
+    abstract public function sendRequest(RequestInterface $request, array $options = []);
+
+    /**
+     * {@inheritdoc}
+     */
+    public function sendRequests(array $requests, array $options = [])
+    {
+        $batchResult    = new BatchResult();
+        $batchException = new BatchException();
+        $batchException->setResult($batchResult);
```

Comment:
Since this is the first case we use the batch result/exception solution, we don't have any reference. But I wonder if we should create both the result and the exception at the beginning. Shouldn't we create the excetpion on demand?